lucene/core/src/java/org/apache/lucene/util/fst/FST.java

```diff
@@ -195,10 +209,9 @@
         posArcsStart = other.posArcsStart();
         arcIdx = other.arcIdx();
         numArcs = other.numArcs();
-        if (nodeFlags() == ARCS_FOR_DIRECT_ADDRESSING) {
-          bitTable = other.bitTable() == null ? null : other.bitTable().copy();
-          firstLabel = other.firstLabel();
-        }
+        bitTableStart = other.bitTableStart;
```

Comment:
I don't see the full code in the diff but I believe all fields should be copied (or cleared, depending on the condition) from other. It may be a bit of an overhead but it'd keep internal state of arcs consistent (and debugging easier). I recall this code was copying just the required fields depending on nodeFlags() - it should clear or copy everything, really.

Reply:
Ok. You've debugged this code for a long time so now you know :)
I'll change that and I'll put a comment to explain.